### PR TITLE
Allow to configure PBKDF

### LINKED ISSUE #152: luks_device: allow to configure number of iterations
##### SUMMARY
Right now it is to possible to set the number of iterations for the key derivation function. A consequence of this is that the integration tests need ~3 minutes to run, since every option / create / update operation takes ~7 seconds.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
luks_device

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #152 (and implements some more).

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
luks_device

### CODE REVIEW

**Comment by felixfontein** [comment]
The luks_device tests now run in < 50 seconds (https://app.shippable.com/github/ansible-collections/community.crypto/runs/843/7/console), as opposed to ~3 minutes (see #152).

**Comment by felixfontein** [comment]
@mulatinho if you have some time, please also take a look at this luks_device PR :-)

File: plugins/modules/luks_device.py [hunk 1]
```diff
@@ -170,6 +170,47 @@
             - "Will only be used on container creation."
         type: str
         version_added: '1.1.0'
+    pbkdf:
+        description:
+            - "This option allows the user to configure the Password-Based Key Derivation
+               Function (PBKDF) used."
+            - "Will only be used on container creation, and when adding keys to an existing
+               container."
```
Comment (Andersson007) [inline plugins/modules/luks_device.py:178]:
```suggestion
            - This option allows the user to configure the Password-Based Key Derivation
               Function (PBKDF) used.
            - Will only be used on container creation, and when adding keys to an existing
               container.
```
I'd avoid unnecessary things. It's just a suggestions, if disagree, feel free to mark as resolved and ignore

**Review by Andersson007 — COMMENTED** [review]
couldn't we cover all the suboptions in CI?

**Comment by felixfontein** [comment]
@Andersson007 the main reason is that not every version of cryptsetup supports all of them. I found a way to make it still work though :)

File: plugins/modules/luks_device.py [hunk 2]
```diff
@@ -170,6 +170,47 @@
...
+            - Will only be used on container creation, and when adding keys to an existing
+              container.
+        type: dict
+        version_added: '1.4.0'
+        suboptions:
+            iteration_time:
+                description:
+                    - "Specify the iteration time used for the PBKDF."
+                    - "Note that this is in B(seconds), not in milliseconds as on the
+                      command line."
```
Comment (Andersson007) [inline plugins/modules/luks_device.py:186]:
```suggestion
                    - Specify the iteration time used for the PBKDF.
                    - Note that this is in B(seconds), not in milliseconds as on the
                      command line.
```

File: plugins/modules/luks_device.py [hunk 3]
```diff
@@ -170,6 +170,47 @@
...
+                    - Mutually exclusive with I(iteration_count).
+                type: float
+            iteration_count:
+                description:
+                    - "Specify the iteration count used for the PBKDF."
```
Comment (Andersson007) [inline plugins/modules/luks_device.py:191]:
```suggestion
                    - Specify the iteration count used for the PBKDF.
```

**Review by Andersson007 — APPROVED** [review]

**Comment by felixfontein** [comment]
@Andersson007 thanks a lot for reviewing this!

**Pull request merged**